### ISSUE TITLE
The marco IS_LOOKUP_NODE creates unused parameters

### ISSUE THREAD

**Issue by Gnnng** [Zilliqa/Zilliqa#156]
Using `IS_LOOKUP_NODE` for source control is fine in general. However, the current use of it interferes with the compile option `-Werror=unused-parameter` (enabled by `-Wextra`) when a parameter is used in one condition but not used in another. Here' s an example from our codebase.

```cpp
~ void Node::LogReceivedDSBlockDetails(const DSBlock& dsblock)
  {
+     //XXX: a workaround to supress -Werror=unused-parameter when not IS_LOOKUP_NODE
+     (void)dsblock;
  #ifdef IS_LOOKUP_NODE
      LOG_MESSAGE2(to_string(m_mediator.m_currentEpochNum).c_str(),
                   "I the lookup node have deserialized the DS Block");
      LOG_MESSAGE2(to_string(m_mediator.m_currentEpochNum).c_str(),
                   "dsblock.GetHeader().GetDifficulty(): "
                       << (int)dsblock.GetHeader().GetDifficulty());
      ...
      LOG_MESSAGE2(to_string(m_mediator.m_currentEpochNum).c_str(),
                   "dsblock.GetHeader().GetLeaderPubKey(): "
                       << dsblock.GetHeader().GetLeaderPubKey());
  #endif // IS_LOOKUP_NODE
  }
```

**Fix**
The workaround looks not clean and is a *workaround*.  This  [page](https://www.jarchitect.com/QACenter/index.php?qa=132&qa_1=conditional-compilation-practices)  provides several options.

**Comment by Gnnng** [comment]
This is blocking #148

**Comment by Gnnng** [comment]
@bb111189 any thoughts?

**Comment by Gnnng** [comment]
There's no good fix for this other than implementing #240. Now we specify `[[gnu::unused]]` anyway to suppress the warnings (see #365)
